boto3/dynamodb/table.py

```diff
@@ -145,13 +145,12 @@ def _flush(self):
             RequestItems={self._table_name: items_to_send}
         )
         unprocessed_items = response['UnprocessedItems']
-
-        if unprocessed_items and unprocessed_items[self._table_name]:
-            # Any unprocessed_items are immediately added to the
-            # next batch we send.
-            self._items_buffer.extend(unprocessed_items[self._table_name])
-        else:
-            self._items_buffer = []
+        if not unprocessed_items:
+            unprocessed_items = {}
+        unprocessed_items = unprocessed_items.get(self._table_name, [])
```

Comment:
Ah my mistake, I see the confusion now, that was lazy on my part. This should either be assigned to a new variable or passed directly to extend.

Reply:
```suggestion
        item_list = unprocessed_items.get(self._table_name, [])
        # Any unprocessed_items are immediately added to the
        # next batch we send.
        self._items_buffer.extend(item_list)
```

Reply:
done!